Makefile

```diff
@@ -305,6 +305,11 @@ ifeq (${ARCH},aarch32)
     endif
 endif
 
+# When building for systems with hardware-assisted coherency, there's no need to
+# use USE_COHERENT_MEM. Require that USE_COHERENT_MEM must be set to 0 too.
+ifeq ($(HW_ASSISTED_COHERENCY)-$(USE_COHERENT_MEM),1-1)
+$(error USE_COHERENT_MEM cannot be enabled with HW_ASSISTED_COHERENCY)
```

Comment:
> I think the message here should probably be: USE_COHERENT_MEM is redundant given HW_ASSISTED_COHERENCY is defined, use one.

Although in effect what you say is true, it sounds a bit vague to me.  Since this is a build error, IMO it is better to give a clear message that the 2 options are incompatible.  Hence the message.